aws/resource_aws_macie2_classification_job_test.go

```diff
@@ -0,0 +1,307 @@
+		ErrorCheck:        testAccErrorCheck(t, macie2.EndpointsID),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsMacieClassificationJobconfigNameGenerated(bucketName, accountID, macie2.JobTypeOneTime),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsMacie2ClassificationJobExists(resourceName, &macie2Output),
+					naming.TestCheckResourceAttrNameGenerated(resourceName, "name"),
+					resource.TestCheckResourceAttr(resourceName, "name_prefix", "terraform-"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAwsMacie2ClassificationJob_NamePrefix(t *testing.T) {
+	var macie2Output macie2.DescribeClassificationJobOutput
+	resourceName := "aws_macie2_classification_job.test"
+	bucketName := os.Getenv("AWS_S3_BUCKET_NAME")
+	accountID := os.Getenv("AWS_ACCOUNT_ID")
+	namePrefix := "tf-acc-test-prefix-"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      nil,
+		ErrorCheck:        testAccErrorCheck(t, macie2.EndpointsID),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsMacieClassificationJobconfigNamePrefix(namePrefix, bucketName, accountID, macie2.JobTypeOneTime),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsMacie2ClassificationJobExists(resourceName, &macie2Output),
+					naming.TestCheckResourceAttrNameFromPrefix(resourceName, "name", namePrefix),
+					resource.TestCheckResourceAttr(resourceName, "name_prefix", namePrefix),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAwsMacie2ClassificationJob_complete(t *testing.T) {
+	var macie2Output macie2.DescribeClassificationJobOutput
+	resourceName := "aws_macie2_classification_job.test"
+	bucketName := os.Getenv("AWS_S3_BUCKET_NAME")
+	accountID := os.Getenv("AWS_ACCOUNT_ID")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      nil,
+		ErrorCheck:        testAccErrorCheck(t, macie2.EndpointsID),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsMacieClassificationJobconfigComplete(bucketName, accountID, macie2.JobStatusRunning),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsMacie2ClassificationJobExists(resourceName, &macie2Output),
+					resource.TestCheckResourceAttr(resourceName, "job_type", macie2.JobTypeScheduled),
+					resource.TestCheckResourceAttr(resourceName, "job_status", macie2.JobStatusRunning),
+					testAccCheckResourceAttrRfc3339(resourceName, "created_at"),
+					testAccCheckResourceAttrRfc3339(resourceName, "last_run_time"),
+				),
+			},
+			{
+				Config: testAccAwsMacieClassificationJobconfigComplete(bucketName, accountID, macie2.JobStatusUserPaused),
```

Comment:
Since most parameters are set to `ForceNew`, we should add a test check to confirm that the resource is *not* recreated when `job_status` is changed. The value of `CreatedAt` can be used to check. e.g.

```go
func testAccCheckAwsMacie2ClassificationJobNotRecreated(i, j *macie2.DescribeClassificationJobOutput) resource.TestCheckFunc {
	return func(s *terraform.State) error {
		if !aws.TimeValue(i.CreatedAt).Equal(aws.TimeValue(j.CreatedAt)) {
			return fmt.Errorf("Macie Classification Job recreated")
		}

		return nil
	}
}
```